### PR TITLE
mgr/cephadm: agent: log response from mgr

### PR DESCRIPTION
**Pull request by adk3798**
Signed-off-by: Adam King <adking@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -3703,7 +3703,11 @@ WantedBy=ceph-{fsid}.target
             url = f'https://{self.target_ip}:{self.target_port}/data'
             try:
                 req = Request(url, data, {'Content-Type': 'application/json'})
-                urlopen(req, context=ssl_ctx)
+                logger.info(f'Sending metadata to mgr at time {datetime.datetime.utcnow().strftime(DATEFMT)}')
+                with urlopen(req, context=ssl_ctx) as response:
+                    response_str = response.read()
+                    response_json = json.loads(response_str)
+                    logger.info(f'Received mgr response: "{response_json["result"]}" at time {datetime.datetime.utcnow().strftime(DATEFMT)}')
```
[the remark this answers is not in the document]
Comment (adk3798) [reply src/cephadm/cephadm:3710]: yeah, that makes more sense here actually. I'll swap over to that

**Review by sebastian-philipp — APPROVED** [review]

File: src/pybind/mgr/cephadm/agent.py [hunk 2]
```diff
@@ -197,7 +201,7 @@ def handle_metadata(self, data: Dict[str, Any]) -> None:
                     f'Got metadata from agent on host {host} with no known counter entry. Starting counter at 1 and requesting new metadata')
                 self.mgr.cache.agent_counter[host] = 1
                 self.mgr.agent_helpers._request_agent_acks({host})
-                return
+                return 'No known counter value. Counter started at 1, new metadata requested'
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/agent.py:204]: (nit) combine the log msg and return value into the same str?

**Review by mgfritch — APPROVED** [review]

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-12-03_09:41:55-orch:cephadm-wip-swagner-testing-2021-12-02-1454-distro-default-smithi/

* https://tracker.ceph.com/issues/49287
* https://tracker.ceph.com/issues/49287

**Comment by sebastian-philipp** [comment]
`AttributeError: module 'mistune' has no attribute 'BlockGrammar'`. I think you need to rebase

**Pull request merged**